homeassistant/components/climate/xs1.py

```diff
@@ -0,0 +1,103 @@
+"""
+Support for XS1 climate devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.xs1/
+"""
+from functools import partial
+import logging
+
+from homeassistant.components.climate import ATTR_TEMPERATURE, ClimateDevice
+from homeassistant.components.xs1 import (
+    ACTUATORS, DOMAIN as COMPONENT_DOMAIN, SENSORS, XS1DeviceEntity)
+
+DEPENDENCIES = ['xs1']
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TEMP = 8
+MAX_TEMP = 25
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the XS1 thermostat platform."""
+    from xs1_api_client.api_constants import ActuatorType
+
+    actuators = hass.data[COMPONENT_DOMAIN][ACTUATORS]
```

Comment:
I understand now that the climate platform is not something the gateway actually supports.

We should remove this platform. It's already possible to use the generic thermostat climate platform for this purpose.

Reply:
What do you mean by "dosen't support"? o.O I used this climate platform for years without any problem (in my own custom component)

Reply:
It's just individual switches and sensors, correct? Not an actual thermostat?

Reply:
The problem is that we shouldn't have more than one platform for the same api representation. If they are are switches and sensors they should be represented by those platforms. If they are part of a thermostat we should only use the climate platform. We have to decide what they are.

Will the switches and sensors always be hooked up to a thermostat or is that just one of many use cases? If thats the case we should let them be switches and sensors and not make a climate platform.

Reply:
But it's not just a switch, you can set the temperature directly on the thermostat using the `actuator` and read the room temperature using the `sensor`. As I understand it this can't be reproduced by using a generic thermostat can it? An `actuator`  is not only a switch, it can be of many different types.

Reply:
I read the code again now. Am I right that the same switch won't show up under two different platforms? It looks like the thermostat switches will only be connected to the climate platform. That's good.

How about the sensors? Will they show up under more than one platform?

Reply:
I don't know why you call it a switch but yes, the "switch" **wont** show under two different platforms, the sensor in it's current implementation **will** show up as normal temperature sensor too.

Reply:
I meant actuator, sorry.

Could we limit the thermostat sensor to the climate platform?

Reply:
Any thoughts on this?
Imo it works out as it is. When the user does not know about the naming convention he will just get a temperature sensor and a climate device. If he knows about it they will be combined in a single sensor (and only for climate devices). I don't think this would mess up anything.

Reply:
But all device sensors will always be set up as sensor entities. Can we try decide already in the component, what devices will be set up under what platforms? Ie, move the check for name to the component, and split sensors between climate and sensor platforms, if there's a name convention.

Reply:
I added the same check to the sensor platform to prevent the sensor entity from showing up in two platforms if it used in a climate entity.